Add TimescaleDB PostgreSQL extension

shusaan opened the pull request:
This PR adds support for the timescaledb extension to the postgres-extensions-containers project.

Closes #58 
## Changes
- Added timescaledb directory with Dockerfile and metadata.hcl
- Updated CI/CD workflows to include timescaledb in the build pipeline

NiccoloFei committed:
feat: add `pgvector` extension and project scaffolding (#3)

This commit introduces the `pgvector` extension image and sets up the initial
project structure. It also includes generic build, test, and update pipelines
to support the future addition of other PostgreSQL extensions.

Closes #2

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>
Signed-off-by: Francesco Canovai <francesco.canovai@enterprisedb.com>
Signed-off-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>
Signed-off-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>
Co-authored-by: Francesco Canovai <francesco.canovai@enterprisedb.com>
Co-authored-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>
Co-authored-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

NiccoloFei committed:
feat: introduce PostGIS extension container image (#5)

PostGIS container image:

- Added a dedicated PostGIS extension image.
- Automatically derives required OS dependencies by diffing the base minimal
  image against the additional libraries installed by PostGIS.
- Added PostGIS-specific Chainsaw tests to validate that all PostGIS-related
  extensions can be successfully installed.

Additional improvements:

- Enabled manual dispatch of builds for a specific extension target.
- Removed `update.yml` in favour of Renovate’s Deb datasource for retrieving
  PGDG packages.
- Converted boilerplate files from JSON to HCL to support Renovate comments and
  provide better customisation of the default Bake target.
- Added support for specifying a custom name for the target container image,
  allowing it to differ from the extension’s name.
- Adapted `pgvector` to above changes.
- Created `BUILD.md` with detailed instructions.
- Created a `Makefile` for a controlled local image building experience.

Closes #4

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>
Signed-off-by: Francesco Canovai <francesco.canovai@enterprisedb.com>
Signed-off-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>
Co-authored-by: Francesco Canovai <francesco.canovai@enterprisedb.com>
Co-authored-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>
Signed-off-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

mnencia committed:
chore: improve Makefile (#12)

When the shell used for `sh` is `dash`, which is common on Debian and
its derivatives, the output of the Makefile is incorrect. This patch
resolves the issue by using `bash` as the shell when it's available.

Additionally, I refactored the `push` target to reuse the code that
executes the push command for a specific directory.

Lastly, the default action when you run `make` without any arguments
is now to build all the extensions.

Signed-off-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>
Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>
Co-authored-by: Niccolò Fei <niccolo.fei@enterprisedb.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

shusaan committed:
feat: add TimescaleDB to workflow options and update filters

Signed-off-by: Husn E Rabbi <shussan@gmail.com>

shusaan committed:
fix: remove redundant postgresql section in README.md

Signed-off-by: Husn E Rabbi <shussan@gmail.com>

sxd committed:
chore: all the GitHub actions update into one PR (#33)

We create only one PR for all the GitHub actions updates.

Closes #32

Signed-off-by: Jonathan Gonzalez V. <jonathan.abdiel@gmail.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

renovate[bot] committed:
chore(deps): update all github action (#34)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

renovate[bot] committed:
chore(deps): update actions/checkout action to v6 (#35)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

shusaan committed:
feat: add `pgaudit` extension (#1)

Closes #22

Signed-off-by: Husn E Rabbi <shussan@gmail.com>
Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>
Signed-off-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>
Co-authored-by: Niccolò Fei <niccolo.fei@enterprisedb.com>
Co-authored-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

gbartolini committed:
docs: licensing and governance policies (#38)

Closes #16

Signed-off-by: Husn E Rabbi <shussan@gmail.com>

NiccoloFei committed:
feat: add `pgmajor` and `os` labels and annotations (#43)

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

NiccoloFei committed:
ci: properly handle an empty strategy matrix (#44)

Closes #39

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

GabriFedi97 committed:
ci: automate the update of OS libraries (#45)

Add automation to verify if we need to rebuild container images
due to OS dependencies updates being available.
This PR adds a dagger module exposing the commands to perform
dependency updates checks only.
A set of files will be maintained under the `system-libs` folder for each
extension requiring OS deps, which will keep track of dependencies updates.
The related workflow will check if any of these files change in any of the
extension folders, and it will open a dedicated PR for each extension.
Each extension has to explicitly set the metadata field `auto_update_os_libs = true`
in the `metadata.hcl` file in order to be part of this mechanism.

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>
Signed-off-by: Gabriele Fedi <gabriele.fedi@enterprisedb.com>
Signed-off-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>
Co-authored-by: Niccolò Fei <niccolo.fei@enterprisedb.com>
Co-authored-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

renovate[bot] committed:
chore(deps): update actions/checkout action to v6 (#50)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

renovate[bot] committed:
chore(deps): update all github action (#37)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

renovate[bot] committed:
chore(deps): update peter-evans/create-pull-request action to v8 (#52)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

mnencia committed:
docs: add image labels and annotations documentation (#46)

Add comprehensive documentation section describing OCI-compliant labels
and CloudNativePG-specific annotations available in extension images.

This documents the labels added in #43, including pgmajor and OS
distribution metadata, plus standard OCI image specification labels.

Relates to #43

Signed-off-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

NiccoloFei committed:
ci: allow creating PRs for OS libraries changes (#51)

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

gbartolini committed:
chore(github): implement 'New Extension Proposal' issue template (#48)

Introduces a required issue template for requesting the inclusion of new
PostgreSQL extensions into the containers project.

This template ensures all necessary information is provided upfront,
specifically requiring:

- Confirmation that no duplicate request exists.
- Project metadata (name, URLs, description, license).
- Mandatory confirmation that the license is compliant.
- Identification of known dependent extensions.
- Explicit commitment from the proposer to act as the Component Owner
   and maintain the extension image in compliance with project governance.

Closes #41
Signed-off-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>
Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>
Signed-off-by: Matteo <matteo.biagetti@enterprisedb.com>
Co-authored-by: Niccolò Fei <niccolo.fei@enterprisedb.com>
Co-authored-by: Matteo <matteo.biagetti@enterprisedb.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

cnpg-bot committed:
chore: update postgis OS libraries (#57)

Signed-off-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: extension-os-libs-updater <extension-os-libs-updater@users.noreply.github.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

shusaan committed:
add timescaledb in README

Signed-off-by: Husn E Rabbi <shussan@gmail.com>

NiccoloFei committed:
feat: automate generation of chainsaw testing values (#56)

Implement a new dagger command to automate the creation
of Chainsaw's testing `values.yaml` for a target extension.

Closes #47

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>
Signed-off-by: Gabriele Fedi <gabriele.fedi@enterprisedb.com>
Co-authored-by: Gabriele Fedi <gabriele.fedi@enterprisedb.com>
Signed-off-by: Husn E Rabbi <shussan@gmail.com>

shusaan committed:
fix: restore *shared to postgis

Signed-off-by: Husn E Rabbi <shussan@gmail.com>

shusaan commented on .github/workflows/bake.yml:
```diff
@@ -48,6 +49,8 @@ jobs:
               - *shared
             postgis:
               - 'postgis/**'
+            timescaledb:
```
fix: restore shared for postgis

phippg commented:
I think you don't necessarily need to build from source, there are Debian packages that include only the open source parts of TimescaleDB (e.g. https://packagecloud.io/timescale/timescaledb/packages/debian/trixie/timescaledb-2-oss-postgresql-18_2.24.0~debian13-1801_amd64.deb?distro_version_id=221). The listed files don't include any `timescaledb-tsl-*.so` binaries.

According to the TimescaleDB Toolkit repository it is fully TSL-licensed unfortunately. If the CNCF license requirements are set in stone for the addition of extension images to this repository, I won't be able to use ones for TimescaleDB (but I do understand that there are valid reasons for them).

shusaan commented:
> I think you don't necessarily need to build from source, there are Debian packages that include only the open source parts of TimescaleDB (e.g. https://packagecloud.io/timescale/timescaledb/packages/debian/trixie/timescaledb-2-oss-postgresql-18_2.24.0~debian13-1801_amd64.deb?distro_version_id=221). The listed files don't include any `timescaledb-tsl-*.so` binaries.
> 
> According to the TimescaleDB Toolkit repository it is fully TSL-licensed unfortunately. If the CNCF license requirements are set in stone for the addition of extension images to this repository, I won't be able to use ones for TimescaleDB (but I do understand that there are valid reasons for them).

ok thank you Let me update PR

shusaan committed:
feat(timescaledb): switch to Apache 2.0 licensed OSS edition

- Replace timescaledb-2-postgresql with timescaledb-2-oss-postgresql package
- Remove timescaledb-2-loader-postgresql dependency
- Update Dockerfile to use OSS edition for CNCF licensing compliance
- Update README to clarify Apache 2.0 licensed components only
- Update metadata.hcl with correct OSS package names and versions
- Bump TimescaleDB version from 2.23.1 to 2.24.0 for both bookworm and trixie
- Ensures compliance with CNCF licensing requirements by excluding TSL-licensed features

Signed-off-by: Husn E Rabbi <shussan@gmail.com>
